src/main/java/org/apache/pirk/utils/SystemConfiguration.java

```diff
@@ -160,4 +139,78 @@ public static void resetProperties()
     clearProperties();
     initialize();
   }
+
+  /**
+   * Loads the properties from local properties file in the specified directory
+   * <p>
+   * Only files ending in '.properties' will be loaded
+   */
+  public static void loadPropsFromDir(String dirName)
+  {
+    File[] directoryListing = new File(dirName).listFiles(new FilenameFilter()
+    {
+      @Override
+      public boolean accept(File dir, String name)
+      {
+        return name.endsWith(".properties");
+      }
+    });
+
+    if (directoryListing != null)
+    {
+      for (File file : directoryListing)
+      {
+        loadPropsFromFile(file);
+      }
+    }
+  }
+
+  /**
+   * Loads the properties from the specified file
+   */
+  public static void loadPropsFromFile(File file)
+  {
+    if (file.exists())
+    {
+      try (InputStream stream = new FileInputStream(file);)
+      {
+        logger.info("Loading properties file '" + file.getAbsolutePath() + "'");
+        props.load(stream);
+        stream.close();
+      } catch (IOException e)
+      {
+        logger.error("Problem loading properties file '" + file.getAbsolutePath() + "'");
+        e.printStackTrace();
+      }
+    }
+    else
+    {
+      logger.warn("Properties file does not exist: '" + file.getAbsolutePath() + "'");
+    }
+  }
+
+  /**
+   * Loads the properties from the specified file on the classpath
+   */
+  public static void loadPropsFromStream(String name)
+  {
+    try
+    {
+      InputStream stream = SystemConfiguration.class.getClassLoader().getResourceAsStream(name);
+      if (stream != null)
+      {
+        logger.info("Loading file '" + name + "'");
+        props.load(stream);
+        stream.close();
```

Comment:
Can use try-with-resources to ensure the stream is closed even if the load causes an exception.